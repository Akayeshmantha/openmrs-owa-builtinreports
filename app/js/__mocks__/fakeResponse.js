

export const FAKE_RESPONSE = {
    dataSets: [
        {
            metadata: {
                columns: [
                    {
                        name: "username",
                        label: "username"
                    },
                    {
                        name: "date_created",
                        label: "date_created"
                    }
                ]
            },
            rows: [
                {
                    date_created: "2010-04-26T13:25:00.000+0530",
                    username: "daemon"
                },
                {
                    date_created: "2017-06-08T21:37:18.000+0530",
                    username: "clerk"
                },
                {
                    date_created: "2017-07-08T21:37:18.000+0530",
                    username: "nurse"
                },
                {
                    date_created: "2017-07-08T21:37:19.000+0530",
                    username: "doctor"
                },
                {
                    date_created: "2017-07-08T21:37:18.000+0530",
                    username: "sysadmin"
                }
            ]
        }
    ],
    definition: {
        name: 'test report'
    }

};