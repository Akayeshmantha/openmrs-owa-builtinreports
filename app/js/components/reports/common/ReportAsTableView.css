@import url('https://fonts.googleapis.com/css?family=Crete+Round|Yantramanav');


.react-grid-HeaderCell{
      background-color: #ffe6aa;
      font-family: 'Yantramanav', sans-serif;
      text-align: center;
}

.react-grid-Cell{
      text-align: center;
}