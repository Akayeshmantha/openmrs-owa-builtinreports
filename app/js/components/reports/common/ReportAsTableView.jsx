import React, { Component } from 'react';
import { ApiHelper } from '../../../helpers/apiHelper';
import * as ReportConstants from '../../../helpers/ReportConstants';
import './ReportAsTableView.css';

/**
 * Display results of a report as a table
 */
class ReportAsTableView extends Component {

    constructor(props) {
        super();
        this.state = {
            report: {
                definition: {
                    name: ''
                }
            },
            reportColumnNames: Array(),
            reportRowData: Array()
        };
        this.getReportUUID = this.getReportUUID.bind(this);
        this.getReportParameter = this.getReportParameter.bind(this);
        this.resolveResponse = this.resolveResponse.bind(this);
    }

    getReportUUID() {
        return this.props.reportUUID;
    }

    getReportParameter() {
        return this.props.reportParameters;
    }

    resolveResponse(data) {
        this.setState({ report: data });
        this.setState({ reportColumnNames: data.dataSets[0].metadata.columns });
        this.setState({ reportRowData: data.dataSets[0].rows });
    }


    componentDidMount() {

        new ApiHelper().post(ReportConstants.REPORT_REQUEST + this.getReportUUID(), this.getReportParameter())
            .then((response) => {
                this.resolveResponse(response);
            });
    }


    render() {
        return (
            <div>
                <h1>
                    {this.state.report.definition.name}
                </h1>

                <table className="reportTable">
                    <thead>
                        <tr>
                            {
                                this.state.reportColumnNames.map(function (element) {
                                    return (
                                        <th key={element.name}>{element.label}</th>
                                    )
                                })
                            }
                        </tr>
                    </thead>
                    <tbody>
                        {
                            this.state.reportRowData.map(function (rowObj, index) {

                                return (
                                    <tr key={index}>
                                        {
                                            this.state.reportColumnNames.map(function (element) {

                                                return (<td key={element.name}>{rowObj[element.name]}</td>)

                                            })

                                        }
                                    </tr>
                                )

                            }, this)
                        }
                    </tbody>

                </table>
            </div>
        );
    }

}

export default ReportAsTableView;