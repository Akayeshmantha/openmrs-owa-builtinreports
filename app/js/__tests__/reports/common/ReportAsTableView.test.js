import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import { fakeRequestLibrary } from '../../../__mocks__/fakeRequestLibrary';
import ReportAsTableView from '../../../components/reports/common/ReportAsTableView';

const SERVER_URL = "http://localhost:8080/openmrs/ws/rest/v1/reportingrest/reportdata/e451ae04-4881-11e7-a919-92ebcb67fe33";
const FAKE_RESPONSE = {
    dataSets: [
        {
            metadata: {
                columns: [
                    {
                        name: "username",
                        label: "username"
                    },
                    {
                        name: "date_created",
                        label: "date_created"
                    }
                ]
            },
            rows: [
                {
                    date_created: "2010-04-26T13:25:00.000+0530",
                    username: "daemon"
                },
                {
                    date_created: "2017-06-08T21:37:18.000+0530",
                    username: "clerk"
                },
                {
                    date_created: "2017-07-08T21:37:18.000+0530",
                    username: "nurse"
                },
                {
                    date_created: "2017-07-08T21:37:19.000+0530",
                    username: "doctor"
                },
                {
                    date_created: "2017-07-08T21:37:18.000+0530",
                    username: "sysadmin"
                }
            ]
        }
    ]

};

describe('<ReportAsTableView /> ', () => {
    it('renders correctly with success data received from server', () => {
        const params = {
            "startDate": "2017-05-05",
            "endDate": "2017-10-05"
        };

      
        var rendered = renderer.create(
            <ReportAsTableView reportUUID="e451ae04-4881-11e7-a919-92ebcb67fe33"
                reportParameters={params}
                fetchData={fakeRequestLibrary('www.openmrs-fake-server.org', {}, true, FAKE_RESPONSE)} />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});