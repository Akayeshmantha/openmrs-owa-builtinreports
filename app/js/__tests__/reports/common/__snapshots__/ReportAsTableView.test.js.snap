// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportAsTableView />  renders correctly with success data received from server 1`] = `
<div
  style={
    Object {
      "border": "1px solid black",
    }
  }
>
  <div
    className="NotFoundWrapper"
  >
    <div
      className="attentionSign"
    >
      <img
        src="./warning.png"
        width="300"
      />
    </div>
    <div>
      No Data found 
      <span>
        for 
        Report Table
      </span>
    </div>
  </div>
</div>
`;
